pkg/registry/apis/provisioning/test.go

```diff
@@ -89,22 +89,31 @@ func (s *testConnector) Connect(ctx context.Context, name string, opts runtime.O
 				// HACK: Set the name and namespace if not set so that the temporary repository can be created
 				// This can be removed once we deprecate legacy secrets is deprecated or we use InLineSecureValues as we
 				// use the same field and repository name to detect which one to use.
-				if cfg.GetName() == "" {
-					if name == "new" {
-						// HACK: frontend is passing a "new" we need to remove the hack there as well
-						// Otherwise creation will fail as `new` is a reserved word. Not relevant here as we only "test"
-						name = "hack-on-hack-for-new"
+				if name == "new" {
+					// HACK: frontend is passing a "new" we need to remove the hack there as well
+					// Otherwise creation will fail as `new` is a reserved word. Not relevant here as we only "test"
+					name = "hack-on-hack-for-new"
+				} else {
+					// Copy previous secure values if they exist
```

Comment:
I think the bug with the webhook error is here. We mark the repository as healthy when calling this endpoint.